core/remote/index.js

```diff
@@ -404,6 +408,32 @@ class Remote /*:: implements Reader, Writer */ {
     return this.remoteCozy.diskUsage()
   }
 
+  async usesFlatDomains() /*: Promise<boolean> */ {
+    let { flatSubdomains } = this.config.capabilities
+    if (flatSubdomains == null) {
+      // XXX: Temporary hack to avoid requesting more permissions to know if
+      // the Cozy is using flat or nested subdomains.
+      //
+      // We try to request the nested subdomains version of the settings app.
+      // If we get an SSL error, it means the stack is configured with flat
+      // subdomains.
+      try {
+        const nestedDomain = `settings.${this.config.cozyUrl.split('//')[1]}`
+        await dns.lookup(nestedDomain, 0)
+        flatSubdomains = false
+      } catch (err) {
+        if (err.syscall === 'getaddrinfo' && err.code === 'ENOTFOUND') {
+          flatSubdomains = true
+        } else {
+          throw err
+        }
+      }
+
+      this.config.capabilities = { flatSubdomains }
```

Comment:
[nit] I'm not a fan of using `set` to make the assignation perform a persist

Reply:
It's kind of a hidden behavior yes. But that would mean changing the whole config module.
It'll probably happen someday :).